### PR TITLE
SW-3850 / SW-3851 Add Planting Site details' zone/subzone views with bread crumbs

### PR DESCRIPTION
**Pull request by karthikbtf**
- extract out a common page framework with crumbs (from Observations DetailsPage.tsx)
- add search selectors for zone/subzone/monitoring-plots in planting site details
- added tables/renderers
- the zone/subzone content was very similar so added into one PR also to see refactor/componentizing opportunities
- fixed a monitoring plots population error in planting site observation results aggregation selector

Terraware App 2023-07-12 17-26-12

Terraware App 2023-07-12 16-24-17

### CODE REVIEW

File: src/components/BreadCrumbs/index.tsx [hunk 1]
```diff
@@ -47,3 +51,45 @@ export default function BreadCrumbs({ hierarchical, crumbs }: BreadCrumbsProps):
...
+type PageProps = {
+  crumbs: Crumb[];
+  title: string;
+  children: React.ReactNode;
+};
+
+/**
+ * A generic page structure with bread crumbs, title, header wrapper and instantiated children.
+ */
+export function Page({ crumbs, title, children }: PageProps): JSX.Element {
```
Comment (karthikbtf) [inline src/components/BreadCrumbs/index.tsx:64]: moved here from DetailsPage.tsx for reuse

File: src/components/PlantingSites/PlantingSiteDetailsTable.tsx [hunk 2]
```diff
@@ -67,7 +66,6 @@ const columns = (): TableColumnType[] => [
 ];
 
 export default function PlantingSiteDetailsTable({ data, plantingSite }: PlantingSiteDetailsTableProps): JSX.Element {
-  const { activeLocale } = useLocalization();
```
Comment (karthikbtf) [inline src/components/PlantingSites/PlantingSiteDetailsTable.tsx:70]: was unused, removed it

File: src/components/PlantingSites/BoundariesAndZones.tsx [hunk 3]
```diff
@@ -29,7 +29,9 @@ export default function BoundariesAndZones({ plantingSite }: BoundariesAndZonesP
   const { isMobile } = useDeviceInfo();
   const theme = useTheme();
 
-  const data = useAppSelector((state) => searchPlantingSiteZones(state, plantingSite.id, view === 'map' ? '' : search));
+  const data = useAppSelector((state) =>
+    searchPlantingSiteZones(state, plantingSite.id, view === 'map' ? '' : search.trim())
```
Comment (karthikbtf) [inline src/components/PlantingSites/BoundariesAndZones.tsx:33]: just an optimization, remove leading/trailing spaces

File: src/components/Observations/common/DetailsPage.tsx [hunk 4]
```diff
@@ -88,30 +82,8 @@ export default function DetailsPage({
...
-          <Grid item xs={12}>
-            <PageSnackbar />
-          </Grid>
-        </Grid>
-      </PageHeaderWrapper>
-      <Grid container ref={contentRef}>
-        {children}
-      </Grid>
-    </TfMain>
+    <Page crumbs={crumbs} title={title}>
```
Comment (karthikbtf) [inline src/components/Observations/common/DetailsPage.tsx:85]: use the extracted component

File: src/components/PlantingSites/PlantingSiteSubzoneView.tsx [hunk 5]
```diff
@@ -1,9 +1,150 @@
...
+  }
+
+  if (!plantingZone) {
+    history.push(APP_PATHS.PLANTING_SITES_VIEW.replace(':plantingSiteId', plantingSiteId));
+  }
+
+  if (!plantingZone?.plantingSubzones.length) {
+    history.push(
+      APP_PATHS.PLANTING_SITES_ZONE_VIEW.replace(':plantingSiteId', plantingSiteId).replace(':zoneId', zoneId)
+    );
```
Comment (karthikbtf) [inline src/components/PlantingSites/PlantingSiteSubzoneView.tsx:85]: redirect to correct page if data is missing

File: src/components/PlantingSites/PlantingSiteZoneView.tsx [hunk 6]
```diff
@@ -1,9 +1,134 @@
...
+    [search]
+  );
+
+  if (!plantingSite) {
+    history.push(APP_PATHS.PLANTING_SITES);
+  }
+
+  if (!plantingZone) {
+    history.push(APP_PATHS.PLANTING_SITES_VIEW.replace(':plantingSiteId', plantingSiteId));
+  }
```
Comment (karthikbtf) [inline src/components/PlantingSites/PlantingSiteZoneView.tsx:72]: redirect on missing data

File: src/components/PlantingSites/index.tsx [hunk 7]
```diff
@@ -41,13 +47,31 @@ export function PlantingSitesWrapper({ reloadTracking }: PlantingSitesProps): JS
...
       dispatch(requestPlantingSiteObservationsResults(selectedOrganization.id, siteId));
     }
   }, [dispatch, selectedOrganization.id, plantingSiteId, trackingV2]);
 
+  // show spinner while initializing data
+  if (
+    (observationsResults === undefined && !observationsResultsError) ||
+    (plantingSites === undefined && !plantingSitesError)
+  ) {
+    return <CircularProgress sx={{ margin: 'auto' }} />;
```
Comment (karthikbtf) [inline src/components/PlantingSites/index.tsx:72]: lifted this from Observations.. basically spin while we are loading data. this is needed so subsequent views can do error handling with selected data (at this point data was already fetched)

File: src/redux/features/observations/plantingSiteDetailsSelectors.ts [hunk 8]
```diff
@@ -94,15 +94,9 @@ export const selectPlantingSiteZones = createSelector(
           plantingSubzones: zone.plantingSubzones.map((sz) => ({
             ...sz,
             // generate monitoring plots from book-keeping data
-            monitoringPlots: zone.plantingSubzones
```
Comment (karthikbtf) [inline src/redux/features/observations/plantingSiteDetailsSelectors.ts:97]: this was the bug, populating monitoring plots for all subzones (note that the context here is specific to a subzone, we only need to populate monitoring plots for the subzone in context)

**Review by charlie-tf — APPROVED** [review]
Looks good to me

**Comment by charlie-tf** [comment]
Vercel works great 👍

**Pull request merged**